docs/ros-snap.md

```diff
@@ -0,0 +1,355 @@
+        message.data = stream.str();
+
+        ROS_INFO("%s", message.data.c_str());
+
+        publisher.publish(message);
+
+        ros::spinOnce();
+
+        loopRate.sleep();
+    }
+
+    return 0;
+}
+```
+
+The "talker" is now complete.
+
+
+### The Python "listener"
+
+First, create the package:
+
+    $ catkin_create_pkg listener rospy std_msgs
+
+Now make sure the `package.xml` is setup correctly (comments removed for
+brevity):
+
+```xml
+<?xml version="1.0"?>
+<package>
+  <name>listener</name>
+  <version>0.0.0</version>
+  <description>The listener package</description>
+  <maintainer email="ubuntu@todo.todo">ubuntu</maintainer>
+  <license>TODO</license>
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>rospy</build_depend>
+  <build_depend>std_msgs</build_depend>
+  <run_depend>rospy</run_depend>
+  <run_depend>std_msgs</run_depend>
+</package>
+```
+
+Also make sure the `CMakeLists.txt` is setup correctly. Again, it's important to remember the install rules:
+
+```cmake
+cmake_minimum_required(VERSION 2.8.3)
+project(listener)
+
+find_package(catkin REQUIRED COMPONENTS
+  rospy
+  std_msgs
+)
+
+catkin_package()
+
+install(PROGRAMS
+  scripts/listener_node
+  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+)
+
+install(FILES
+  talk_and_listen.launch
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
+```
+
+Now for `scripts/listener_node`, we have this:
+
+```python
+#!/usr/bin/env python
+
+import rospy
+from std_msgs.msg import String
+
+def callback(data):
+    rospy.loginfo('I heard %s', data.data)
+
+def listener():
+    rospy.init_node('listener')
+
+    rospy.Subscriber('babble', String, callback)
+
+    rospy.spin()
+
+if __name__ == '__main__':
+    listener()
+```
+
+Note that the listener subscribes to the `babble` topic, but the talker
+publishes on `chatter`. We need to make sure we account for that in
+`talk_and_listen.launch`:
+
+```xml
+<launch>
+    <node name = "talker" pkg = "talker" type = "talker_node"
+          output = "screen" />
+
+    <node name = "listener" pkg = "listener" type = "listener_node"
+          output = "screen">
+        <remap from = "babble" to = "chatter" />
+    </node>
+</launch>
+```
+
+The "listener" is now complete.
+
+
+### Verify functionality
+
+You can verify that everything works by getting in the workspace root and
+running:
+
+    $ catkin_make
+    $ catkin_make install
+    $ roslaunch listener talk_and_listen.launch
+
+You should see them communicating, looking something like this:
+
+    ...
+    [INFO] [WallTime: 1449512660.316140] I heard Hello world 2
+    [ INFO] [1449512660.415941529]: Hello world 3
+    [INFO] [WallTime: 1449512660.416330] I heard Hello world 3
+    [ INFO] [1449512660.515954882]: Hello world 4
+    [INFO] [WallTime: 1449512660.516307] I heard Hello world 4
+    [ INFO] [1449512660.615954473]: Hello world 5
+    [INFO] [WallTime: 1449512660.616306] I heard Hello world 5
+    ...
+
+### Put it all in a .snap
+
```

Comment:
Maybe add half a sentence about what `snapcraft.yaml` is, what snapcraft aims to do - just so people who get directed to this specific article have a little bit of context.